### PR TITLE
H-3590: Allow all CI to pass when opening PRs from a fork

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

When opening PRs from a fork some CI cannot pass because they require authentication. To avoid this issue, this PR disables these steps (they will still be checked when running the merge-queue)

## 🔍 What does this change?

- Avoid uploading the cache for Build steps
- drive-by: Don't upload the Build results as artifact anymore, it's not used
- Use `checkout` action for Benchmark CI instead of manual commands
- Don't post the Benchmark-result as comment (the action runs from a fork so we cannot use `GH_TOKEN` here) - this can possibly be avoided by adding a `pull_request_target` run but that requires another CI workflow as we must not run the benchmarks using `pull_request_target`.
- Don't upload source maps to Sentry
- Don't attempt to plan a Terraform run

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ on:
 jobs:
   release:
     runs-on: ubuntu-latest
-    if: github.repository == 'hashintel/hash'
+    if: github.repository == github.repository
```
Comment (CiaranMn) [inline .github/workflows/release.yml:12]: Why this change? It will always be true, releasing to npm isn't going to work on forks, and it'll only run on merge to `main` in any case.

Comment (TimDiekmann) [reply .github/workflows/release.yml:12]: Whops, that was a search/replace error (I used `'dashintel/hash/'` previously but replaced it with `GitHub.repository`.
```suggestion
    if: github.repository == 'hashintel/hash'
```

**Review by CiaranMn — APPROVED** [review]
